### Python Fundamentals for Data Analysts

# 4.3: Intro to Pandas

### Table of content

1. Importing CSV files
- usecols = var_list
- index_col = False

2. Print Dataframe
- df_prods.columns
- df_prods.info()
- df_prods.describe()

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os

In [2]:
# Setting path variable to access files
path = r'C:\Users\tsoew\OneDrive\Desktop\InstaCart Basket Analysis'
path

'C:\\Users\\tsoew\\OneDrive\\Desktop\\InstaCart Basket Analysis'

### Import orders.csv file, omitting eval_set column

In [3]:
vars_list = ['order_id', 'user_id', 'order_number', 'order_dow', 'order_hour_of_day', 'days_since_prior_order']

In [4]:
vars_list

['order_id',
 'user_id',
 'order_number',
 'order_dow',
 'order_hour_of_day',
 'days_since_prior_order']

In [5]:
# Import first Dataframe
# Use usecols to select columns
df = pd.read_csv(os.path.join(path, 'Data', 'Original Data', 'orders.csv'), usecols = vars_list)

In [6]:
# Print out the first five rows
df.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

### Import products.csv file

In [7]:
# Import csv file, use index_col = False to include all columns?
df_prods = pd.read_csv(os.path.join(path, 'Data', 'Original Data', 'products.csv'), index_col = False)

**Note**

Column(s) to use as row label(s), denoted either by column labels or column indices. If a sequence of labels or indices is given, MultiIndex will be formed for the row labels.

Note: index_col=False can be used to force pandas to not use the first column as the index, e.g., when you have a malformed file with delimiters at the end of each line.:** Column(s) to use as row label(s), denoted either by column labels or column indices. If a sequence of labels or indices is given, MultiIndex will be formed for the row labels.

Note: index_col=False can be used to force pandas to not use the first column as the index, e.g., when you have a malformed file with delimiters at the end of each line.

In [8]:
df_prods.head()

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

In [9]:
df_prods.tail()

product_id                                product_name  aisle_id  \
49688       49684   Vodka, Triple Distilled, Twist of Vanilla       124   
49689       49685          En Croute Roast Hazelnut Cranberry        42   
49690       49686                            Artisan Baguette       112   
49691       49687  Smartblend Healthy Metabolism Dry Cat Food        41   
49692       49688                      Fresh Foaming Cleanser        73   

       department_id  prices  
49688              5     5.3  
49689              1     3.1  
49690              3     7.8  
49691              8     4.7  
49692             11    13.5

In [10]:
df_prods.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [11]:
df_prods.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49693 entries, 0 to 49692
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   product_id     49693 non-null  int64  
 1   product_name   49677 non-null  object 
 2   aisle_id       49693 non-null  int64  
 3   department_id  49693 non-null  int64  
 4   prices         49693 non-null  float64
dtypes: float64(1), int64(3), object(1)
memory usage: 1.9+ MB


**Note**

1. Data type of department_id is int64

In [12]:
df_prods.describe()

product_id      aisle_id  department_id        prices
count  49693.000000  49693.000000   49693.000000  49693.000000
mean   24844.345139     67.770249      11.728433      9.994136
std    14343.717401     38.316774       5.850282    453.519686
min        1.000000      1.000000       1.000000      1.000000
25%    12423.000000     35.000000       7.000000      4.100000
50%    24845.000000     69.000000      13.000000      7.100000
75%    37265.000000    100.000000      17.000000     11.200000
max    49688.000000    134.000000      21.000000  99999.000000

**Note**
1. product_id column:
    Max = 49688 (is it normal?)
    Average = 24844
    50% = 24845
2. aisle_id column:
    Max = 134.0
3. department_id column:
    Max = 21
4. prices column:
    Avg = 9.99
    Max= 9999 (some issues here!)